packages/core/src/models/MCP.ts

```diff
@@ -0,0 +1,244 @@
+import {
+  MCPGenerateURLParams,
+  MCPCreateMethodResponse,
+  MCPToolkitConfig,
+  MCPAuthOptions,
+  MCPGetServerParams
+} from '../types/mcp.types';
+import {
+  McpCreateResponse,
+  McpListResponse,
+  McpUpdateResponse,
+  McpDeleteResponse,
+  CustomCreateResponse,
+  GenerateURLResponse
+} from '@composio/client/resources/mcp';
+
+/**
+ * MCP (Model Control Protocol) class
+ * Handles MCP server operations
+ */
+export class MCP {
+  private client: ComposioClient;
+
+  constructor(client: ComposioClient) {
+    this.client = client;
+    telemetry.instrument(this);
+  }
+
+  /**
+   * Create a new MCP server
+   * @param {string} name - Unique name for the MCP server
+   * @param {MCPToolkitConfig[]} toolkitConfigs - Array of toolkit configurations
+   * @param {MCPAuthOptions} [authOptions] - Authentication configuration options
+   * @returns {Promise<MCPCreateMethodResponse>} Created server details with instance getter
+   * 
+   * @example
+   * ```typescript
+   * const cfg = await composio.mcp.create(
+   *   "personal-mcp-server",
+   *   [
+   *     {
+   *       toolkit: "GMAIL",
+   *       authConfigId: "ac_sdhkjfhjksdk",
+   *       allowedTools: ["GMAIL_FETCH_EMAILS"],
+   *     },
+   *   ],
+   *   { useManagedAuthByComposio: true },
+   * );
+   * 
+   * // Get server instance
+   * const server = await cfg.getServer({
+   *   connected_account_ids: {
+   *     "gmail": "acc_123",
+   *     "supabase": "acc_456"
+   *   }
+   * });
+   * // OR
+   * const server = await cfg.getServer({
+   *   user_id: "user_123"
+   * });
+   * ```
+   */
+  async create(
+    name: string,
+    toolkitConfigs: MCPToolkitConfig[],
+    authOptions?: MCPAuthOptions
+  ): Promise<MCPCreateMethodResponse> {
+    // Check for unique toolkits
+    const toolkits = toolkitConfigs.map(config => config.toolkit);
+    const uniqueToolkits = new Set(toolkits);
+    if (uniqueToolkits.size !== toolkits.length) {
+      throw new Error('Duplicate toolkits are not allowed. Each toolkit must be unique.');
+    }
+
+    // Create a custom MCP server with the toolkit configurations
+    const mcpServerCreatedResponse: CustomCreateResponse = await this.client.mcp.custom.create({
+      name,
+      toolkits: toolkits,
+      custom_tools: toolkitConfigs.flatMap(config => config.allowedTools),
+      managed_auth_via_composio: authOptions?.useManagedAuthByComposio || false,
+      auth_config_ids: toolkitConfigs.map(config => config.authConfigId)
+    });
+
+    // Add getServer method to response
+    return {
+      ...mcpServerCreatedResponse,
+      toolkits,
+      getServer: async (params: MCPGetServerParams): Promise<GenerateURLResponse> => {
+        // Validate that only one of user_id or connected_account_ids is provided
+        if (params.user_id && params.connected_account_ids) {
+          throw new Error('Cannot specify both user_id and connected_account_ids. Please provide only one.');
```

Comment:
Use zod schemas to parse this and throw `ValidationError`

Reply:
makes sense, fixed it